js/library.js

```diff
@@ -0,0 +1,94 @@
+/* eslint-disable import/extensions */
+import { bookHtml, toggleForm } from './dom.js';
+/* eslint-enable import/extensions */
+
+const myLibrary = [];
+
+// book constructor
+function Book(author, title, pages, status = false) {
+  this.author = author;
+  this.title = title;
+  this.pages = pages;
+  this.status = status;
+  this.id = Math.floor(Math.random() * 10000);
+}
+
+myLibrary.push(new Book('Famous Author', 'The Great Book', 97, true));
+myLibrary.push(new Book('Famous Author Jr.', 'The Even Greater Book', 177, false));
+myLibrary.push(new Book('Not So Famous Author', 'The Not So Great Book', 246, true));
+
+// displays books added
+function render() {
+  const layout = myLibrary.map(book => bookHtml(book)).join(' ');
+  document.getElementById('book-list').innerHTML = layout;
+}
+
+const addBookToLibrary = document.getElementById('submit');
+
+// updating array
+addBookToLibrary.onclick = () => {
+  const author = document.getElementById('author').value;
+  const title = document.getElementById('title').value;
+  const pages = document.getElementById('pages').value;
+  const status = document.getElementById('status').checked;
+
+
+  myLibrary.push(new Book(author, title, pages, status));
+
+  document.getElementById('author').value = '';
+  document.getElementById('title').value = '';
+  document.getElementById('pages').value = '';
+  document.getElementById('status').checked = false;
+  toggleForm('close');
+  render();
+};
```

Comment:
separate DOM from logic addBookToLibrary should be a function that does one thing only ![see this](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/library/book_array_requirement.png)

A good thing:  get input from the form in a fucntion stored in dom.js call fucntion in library set values to the new book constructor then pass to addBookToLibrary function which then just adds the book to library :slightly_smiling_face: